www/js/diary/services.js

```diff
@@ -605,11 +545,10 @@ angular.module('emission.main.diary.services', ['emission.plugin.logger',
                     linkTrips(trip_gj_list[i], trip_gj_list[i+1]);
                 }
                 Logger.log("finished linking trips for list of size "+trip_gj_list.length);
-                if (processedTripList.length != 0 && trip_gj_list.length != 0) {
+                if (lastProcessedTrip && trip_gj_list.length != 0) {
                     // Need to link the entire chain above to the processed data
                     Logger.log("linking unprocessed and processed trip chains");
-                    var last_processed_trip = processedTripList.slice(-1);
```

Comment:
No, this has nothing to do with the reversal.
I just reworked this function `readUnprocessedTrips` to take `lastProcessedTrip` instead of the whole list `processedTripList`, since that's all it needs

Reply:
Right, and you also changed the way that the `lastProcessedTrip` is computed in 
https://github.com/e-mission/e-mission-phone/pull/1000/files#diff-b569b67815ed45115ef60195dc85fb9611c9276a9b28ef51718e5c3cc666aa84R192

So this is more performant, but why was it wrong? Was it not the reverse?

Reply:
It was wrong before because it passed `allTrips` as the processedTripList. `allTrips` was just whatever had been loaded; it might have included unprocessed trips. I didn't notice any adverse effects from this being wrong though.

Is the linking of trips even important in the phone code?

Reply:
Maybe not at this point. But linking it fits the standard data model that we have established, so that if/when we actually use the linking in phone functionality, we don't have to add even more special cases.